### ISSUE TITLE
[Bug]: RTSP + HKSV issues with 2.2B3 - Floodlight Cam T8424 & Eufy Wired Doorbell 2K

### ISSUE THREAD

**Issue by spiezer** [homebridge-eufy-security/plugin#149]
### What happened?

Hello. 

I've been testing out the beta versions and have been having some odd issues with the recent release. I've installed the beta on 3 devices: a pi4, macOS, and Windows 10.  I'm currently using it on my Windows installation since it's the most stable one for the streams.  The only major issue I'm having right now is with motion notifications and RTSP.

RTSP Streams:
It works only on the floodlight. When I attempt to use it on the doorbell I get a connection refused on port 554. On the floodlight, I added the RTSP link to scrypted and it works swimmingly for about 10 minutes then stops working.  I have to restart homebridge to continue the stream. I saw something about maintaining RTSP connection causing issues so if it could cut off every once in a while that would be good.

While on RTSP and using motion analysis, there were no issues parsing movement and getting notifications.  I used Scrypted for this.

Motion Notifications:
With the doorbell, it's inconsistent with the notifications. It seems to depend on the device I'm using but it seems to be completely random. As of right now, prior to another reinstall, it does not seem to work on my iphone but works on my ipad.  Although I do not get notifications on the iphone, it does hit the timeline. So I can scroll through previous times that it did see motion.

I get nothing from the floodlight. I get motion notifications from the app but not anywhere else.  It has not functioned regardless of where I installed homebridge.

Logs: I attached part of homebridge.log that I think is relevant. It's within the 1:40PM range. I also have the associated eufy-log, homebridge, and log-lib files if needed.  

### Device Type

Doorbell (Supported)

### Plugin Version

beta (Best Effort)

### HomeBridge Version

v1.4.x (Supported)

### NodeJS Version

v16 (Supported)

### Operating System

Windows 10+ (Supported)

### Relevant log output

```shell
[8/19/2022, 1:38:52 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetection: true[39m
[8/19/2022, 1:38:52 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetection: true[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetected: false[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceStatusLed: true[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetected: false[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceStatusLed: true[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceLight: false[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetection: true[39m
[8/19/2022, 1:39:55 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetection: true[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetected: false[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceStatusLed: true[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetected: false[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceStatusLed: true[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceLight: false[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetection: true[39m
[8/19/2022, 1:41:38 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetection: true[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetected: false[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceStatusLed: true[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetected: false[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceStatusLed: true[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceLight: false[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetection: true[39m
[8/19/2022, 1:44:07 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetection: true[39m

[8/19/2022, 1:44:18 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell stored new snapshot from cloud in memory.[39m
[8/19/2022, 1:44:18 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell cloud snapshot is most recent one. Storing this for future use.[39m
[8/19/2022, 1:44:18 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage stored new snapshot from cloud in memory.[39m
[8/19/2022, 1:44:18 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage cloud snapshot is most recent one. Storing this for future use.[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetected: false[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceStatusLed: true[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetected: false[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceStatusLed: true[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceLight: false[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetection: true[39m
[8/19/2022, 1:44:57 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetection: true[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetected: false[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceStatusLed: true[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceEnabled: true[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetected: false[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET EventSnapshotsActive: 1[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET HomeKitCameraActive: 1[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET PeriodicSnapshotsActive: 1[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceEnabled: true[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceStatusLed: true[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceLight: false[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mGarage GET DeviceMotionDetection: true[39m
[8/19/2022, 1:45:25 PM] [EufySecurity-2.2.0-beta.3] DEBUG: [36mDoorbell GET DeviceMotionDetection: true[39m
```

**Comment by thieren** [comment]
Hey @spiezer 

There seems to be a lot going on, so to make sure I got everything right, let me recap:

1. you're running on windows right now, but also tested Rpi4 and macOS.
2. You have trouble accessing RTSP on doorbell.
3. Your RTSP stream from the floodlight only runs for 10 minutes (using scrypted).
4. You have an inconsistent experience with motion detection on your doorbell
5. You don't get any motion detection from the floodlight camera but from the eufy app

Let me try to answer them in order:

1. Running on Windows should work fine overall. But HKSV seems to not work properly with windows. I tested this while implementing, but did not find out, why HKSV does not work on windows homebridge instances yet. I'd recommend running on another os if you want to use HKSV. (see [here](https://github.com/homebridge-eufy-security/plugin/issues/6#issuecomment-1207463595)).

2. As far as I know Doorbells do not support RTSP out of the box. So you need to be a bit more specific. Did you use the new [experimental mode](https://github.com/homebridge-eufy-security/plugin/wiki/RC---Beta-Version#experimental-mode) for this? If so, there is no guarantee that this will work fine. (It does work for me, but it's quite experimental - hence the naming ;-) )

3. I don't know a lot about scrypted and maybe I misinterpret your setup here, but I understand that you feed the RTSP-url into scrypted. If that is the case, then there is no interaction with this plugin. scrypted will get the data directly from the device.
  Do you get any stream from scrypted after that? Why don't you use the homebridge plugin from scrypted for this?
  In short: I don't really get what's the issue with this plugin here.

4. That depends on your setup really (see 2.) I found that motion detection is not working on my doorbell when it already streams. So it might be, that scrypted is using the stream sometimes, and therefore motion detection is not working?!
  Sorry but here also it's important to know your exact setup.

5. Is kind of the same way as with 4. I guess.<br>
  Unfortunately your logs don't supply any information, that any motion was detected at any time in this time frame. It only just shows information about device parameters being read. No action whatsoever.

Since there are a bunch of issues, I'd recommend that we go about this one at a time. What is your biggest flaw, you'd like to get rid off?

Also I don't see any mention of HKSV in particular, although it is mentioned in the issue title. Since activated HKSV can have effects on things like motion detection again - I'd need your exact setup (e.g. what's the data-flow for camera, scrypted, homebridge/plugin?)

**Comment by spiezer** [comment]
Hey Thieren,

1. Got it.  I'm going to try switching back over the mac install later. The Windows one is faster but isn't consistently working (live streams seem to break a bit later in the day with HKSV enabled)

2. I did use the new experimental RTSP mode for this.  It seems that the connection is pretty much blocked on the doorbell vs the floodlight.

3. I got the URL from the eufy plugin after I enabled the experimental plugin for RTSP for both devices. Then I used that URL and inserted it into Scrypted. To clarify, the RTSP url from the doorbell never worked and the connection seemed to be blocked. The floodlight one worked for a little bit.  Then when it stopped working, upon rebooting homebridge, it would work for another 10 minutes. 

4. Hmm I think you're right here. I never got the doorbell to work in Scrypted w/ RTSP but I think the motion detection was inconsistent when I had HKSV enabled.  I went ahead and disabled it this morning to see if that resolves any issues.  
Scrypted is a completely different app and not a homebridge plugin. I used it to import the RTSP urls.  Not sure if it accessing the RTSP urls that way caused issues.

5. Not sure about this one since I did get a motion notification from the Eufy app but not from Homekit.

I used an ethernet cable temporarily while i had the mac install but couldnt keep it connected as it's not my main device. I'll go ahead and give things another shot once my adapter comes in.

[log-lib.log](https://github.com/homebridge-eufy-security/plugin/files/9387323/log-lib.log)
[eufy-log.log](https://github.com/homebridge-eufy-security/plugin/files/9387324/eufy-log.log)

**Comment by thieren** [comment]
2. What exact doorbell model do you have? With Battery Doorbell (T8210) it works fine for me.

3. So I did understand correct that you just used the link for scrypted. However I don't see any connection from the plugin to the stream then. The plugin merely reads the RTSP-url from the device. If you do something else after that with it, the plugin has no doing in the things after that.

4. Let me know if disabling HKSV did any good.

5. Kind of hard to say. Your eufy-log.log file just covers a duration of 2 minutes. In that time there seems to be no motion in front of the garage camera at least. But that's a too narrow time frame.

**Comment by spiezer** [comment]
1. The ‎T8200 2k wired doorbell.  Hmm I'll give it another shot with the mac later.
2. Yeah, honestly I'm pretty confused now. It basically stops working until I reboot the homebridge instance (which only contains the eufy plugin) and then it works again.  I'm going to give it another look once I have the ethernet adapter to see if that makes a difference. Maybe multiple streams somehow ran at once and that may have caused it to error out until the homebridge reboot cuts one off?  
3. Yeah the motion detection seems to be working fine again.
4. Oh boy. I'll go ahead and give it another shot later when I have the appropriate setup.
Thanks for your patience!

**Comment by thieren** [comment]
Hey @spiezer 

Any news on this?

**Comment by spiezer** [comment]
Hey Thieren,

I gave it a shot but replaced the doorbell with an Amcrest AD410. Unfortunately the Eufy still seemed a bit unstable and couldn't give me reliable notifications despite being on the mac install.  The actual Eufy doorbell seemed to have but that was probably due to the product and not the plugin.

I tried running RTSP on the floodlight but still had the same issues while on the mac. I think it's a built in limitation since others have mentioned it being a problem. Similar to [here](https://www.reddit.com/r/EufyCam/comments/ju9hr2/eufy_indoor_cam_2k_how_to_enable_live_view_for/) so it's probably something hard coded. (this was the beta feature I enabled on the plugin).

I also tried the motion detection a bit on the floodlight but it wasn't working. Had the same issues I had while I was running on the windows install. No notifications if I had HKSV enabled while it worked when HKSV was not enabled. I'm thinking about returning the doorbell but was waiting for another update before messing around with the floodlight again.
